e2e/rbd.go

```diff
@@ -539,14 +539,26 @@ var _ = Describe("RBD", func() {
 					LabelSelector: fmt.Sprintf("app=%s", app.Name),
 				}
 				filePath := app.Spec.Containers[0].VolumeMounts[0].MountPath + "/test"
-				_, stdErr, err = execCommandInPod(
-					f,
-					fmt.Sprintf("echo 'Hello World' > %s", filePath),
-					app.Namespace,
-					&appOpt)
-				if err != nil || stdErr != "" {
-					e2elog.Failf("failed to write IO, err: %v, stdErr: %v ", err, stdErr)
-				}
+				err = wait.PollImmediate(poll, timeout, func() (bool, error) {
```

Comment:
might be useful to place this in a helper function `writeInPodWithRetry()`?

Reply:
Sure, that should make this less complex and modular.

Reply:
oh! Given this is handling special cases like retry for Read-Only error but not for others, this might not be helpful?